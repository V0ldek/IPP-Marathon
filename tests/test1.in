# Proste drzewo i brak błędów w wejściu

addUser 0 1
addUser 0 2
addUser 0 3

addUser 1 4
addUser 4 5
addUser 5 6

addUser 2 7
addUser 2 8
addUser 2 9

# The tree structure is:
# 0 -> [1; 2; 3]
# 1 -> [4]
# 2 -> [7; 8; 9]
# 4 -> [5]
# 5 -> [6]

addMovie 0 100
addMovie 0 150
addMovie 0 200

addMovie 1 175
addMovie 2 180
addMovie 3 199

addMovie 4 215
addMovie 5 214
addMovie 6 216

addMovie 7 300
addMovie 7 301
addMovie 7 302
addMovie 8 300
addMovie 8 303
addMovie 9 205
addMovie 9 105

# Lists of movies:
# 0: [200; 150; 100]
# 1: [175]
# 2: [180]
# 3: [199]
# 4: [215]
# 5: [214]
# 6: [216]
# 7: [302; 301; 300]
# 8: [303; 300]
# 9: [205; 105]

# Calculating the biggest marathons from leaves up:
# marathon 9 -> [205; 105]
# marathon 8 -> [303; 300]
# marathon 7 -> [302; 301; 300]
# marathon 6 -> [216]
# marathon 3 -> [199]
# marathon 5 -> [216; 214]
# marathon 4 -> [216; 215]
# marathon 2 -> [303; 302; 301; 300; 205; 180]
# marathon 1 -> [216; 215; 175]
# marathon 0 -> [303; 302; 301; 300; 216; 215; 205; 200; 150; 100]

# The biggest marathon you can get for 0 is:
# [303; 302; 301; 300; 216; 215; 205; 200; 150; 100]
marathon 0 100
marathon 0 10
marathon 0 9
marathon 0 8
marathon 0 7
marathon 0 6
marathon 0 5
marathon 0 4
marathon 0 3
marathon 0 2
marathon 0 1
marathon 0 0

# The biggest marathon for 1 is:
# [216; 215; 175]
marathon 1 0
marathon 1 1
marathon 1 2
marathon 1 3
marathon 1 4

# The biggest marathon for 2 is:
# [303; 302; 301; 300; 205; 180]
marathon 2 7
marathon 2 6
marathon 2 5
marathon 2 4
marathon 2 3
marathon 2 2
marathon 2 1
marathon 2 0

# The biggest marathon for 3 is:
# [199]
marathon 3 0
marathon 3 1
marathon 3 2

# The biggest marathon for 4 is:
# [216; 215]
marathon 4 3
marathon 4 2
marathon 4 1
marathon 4 0

# The biggest marathon for 5 is:
# [216; 214]
marathon 5 0
marathon 5 1
marathon 5 2
marathon 5 3

# The biggest marathon for 6 is:
# [216]
marathon 6 0
marathon 6 1
marathon 6 2

# The biggest marathon for 7 is:
# [302; 301; 300]
marathon 7 3
marathon 7 2
marathon 7 1
marathon 7 0

# The biggest marathon for 8 is:
# [303; 300]
marathon 8 0
marathon 8 1
marathon 8 2
marathon 8 3

# The biggest marathon for 9 is:
# [205; 105]
marathon 9 3
marathon 9 2
marathon 9 1
marathon 9 0

