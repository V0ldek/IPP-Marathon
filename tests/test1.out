OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
303 302 301 300 216 215 205 200 150 100
303 302 301 300 216 215 205 200 150 100
303 302 301 300 216 215 205 200 150
303 302 301 300 216 215 205 200
303 302 301 300 216 215 205
303 302 301 300 216 215
303 302 301 300 216
303 302 301 300
303 302 301
303 302
303
NONE
NONE
216
216 215
216 215 175
216 215 175
303 302 301 300 205 180
303 302 301 300 205 180
303 302 301 300 205
303 302 301 300
303 302 301
303 302
303
NONE
NONE
199
199
216 215
216 215
216
NONE
NONE
216
216 214
216 214
NONE
216
216
302 301 300
302 301
302
NONE
NONE
303
303 300
303 300
205 105
205 105
205
NONE
